index.html

```diff
@@ -1884,50 +1909,72 @@ <h2>The <dfn>NDEFRecordType</dfn> string</h2>
       <td><dfn>"`empty`"</dfn></td>
       <td><i>unused</i></td>
       <td><i>unused</i></td>
-      <td><a>Empty record</a></td>
+      <td><a>Empty record</a> (TNF=0)</td>
     </tr>
     <tr>
       <td><dfn>"`text`"</dfn></td>
       <td><i>unused</i></td>
       <td>{{DOMString}}</td>
-      <td><a>Well-known type</a> record with type "`T`"</td>
+      <td><a>Well-known type</a> (TNF=1) record with TYPE="`T`"</td>
     </tr>
     <tr>
       <td><dfn>"`url`"</dfn></td>
       <td><i>unused</i></td>
       <td>{{DOMString}}</td>
-      <td><a>Well-known type</a> record with type "`U`"</td>
+      <td><a>Well-known type</a> (TNF=1) record with TYPE="`U`"</td>
     </tr>
     <tr>
       <td><dfn>"`smart-poster`"</dfn></td>
       <td><i>unused</i></td>
       <td>{{NDEFMessage}}</td>
-      <td><a>Well-known type</a> record with type "`Sp`"</td>
+      <td><a>Well-known type</a> (TNF=1) record with TYPE="`Sp`"</td>
+    </tr>
+    <tr>
+      <td><dfn>"`absolute-url`"</dfn></td>
+      <td><i>unused</i></td>
+      <td>{{DOMString}} <i>url</i></td>
+      <td><a>Absolute-URL record</a> (TNF=2) with TYPE=<i>url</i></td>
     </tr>
     <tr>
       <td><dfn>"`json`"</dfn></td>
       <td><a>JSON MIME type</a></td>
       <td>
         [= JSON type =]
       </td>
-      <td><a>MIME type record</a> with type equal to
-        <a>MIME type</a>.
+      <td><a>MIME type record</a> (TNF=2) with TYPE= [= JSON type =].
       </td>
     </tr>
     <tr>
       <td><dfn>"`opaque`"</dfn></td>
-      <td><a>MIME type</a></td>
+      <td>[= MIME type =]</a></td>
       <td>{{ArrayBuffer}}</td>
-      <td><a>MIME type record</a></td>
+      <td><a>MIME type record</a> (TNF=3) with TYPE= [= MIME type =]</td>
     </tr>
     <tr>
-      <td><a>external type</a></td>
+      <td><i>[= external type =]</i></td>
+      <td><i>unused</i></td>
+      <td>
+        {{ArrayBuffer}} or any<br>
+        <a>typed array type</a> or<br>
+        {{NDEFMessage}}
+      </td>
+      <td><a>External type record</a> (TNF=4) with TYPE= [= external type =]</td>
+    </tr>
+    <tr>
+      <td><i>[= local type =]</i></td>
       <td><i>unused</i></td>
       <td>
-        {{ArrayBuffer}} or<br>
-        any <a>typed array type</a>
+        {{ArrayBuffer}} or any<br>
+        <a>typed array type</a>
       </td>
-      <td><a>External type record</a></td>
+      <td><a>External type record</a> (TNF=4) with TYPE= [= local type =] inside
```

Comment:
From what I read at https://w3c.github.io/web-nfc/#well-known-type-records-tnf-1, it seems to me that a local type record is always saved as a well-known record (TNF=1)?

Reply:
Local types are defined at TNF=1 like this

> NFC Forum Local Types are available for use within the context of another record.

so nothing prevents apps from using them in embedded message in an external type record. 
This is the liberal interpretation of the NFC spec.

However,  the conservative interpretation would agree with you, but that means apps cannot use local types, because they are only defined for "Sp" in TNF=1.

Reply:
Local types are also used for handover.
I fail to decode the initial intent with these local types, but it seems more and more that in actual use they are under well-known types, indeed.

Let's simplify things and I remove them from both sides of data mapping. When we get use cases, we can add them back later.

Reply:
Its easy to explain:

 They are basically external types, but it would be foolish to use up space within a record for re-adding the URL that is required in the external type name, so they avoid adding that, but restricting all types to start with lowercase or number. All standardized well known types start with uppercase so there is no conflict

external type = custom type
local type = custom type scoped within a custom type, thus can avoid the namespacing

A custom type of name say "kenny" can be used in multiple external type records but in that case they are not considered the same type (they are local types, not global) and behavior can differ.

Reply:
I meant to ask whether a local type record is saved as TNF=1 in an nfc tag.

Another question, for the NDEFMessage acting as a payload, should all of its records be local type records?  Or the payload NDEFMessage is just to be a normal NDEFMessage?

Reply:
No, just a normal NDEF message, which may contain local types

Reply:
No, only global types start with upper case, and all local types (including the standard ones) start with lowercase or number.

Local types are defined in the context of WKT, with the mention that they can be used also in other context and with other meaning. The only such place is external records. But that doesn't mean they are basically external types, they are just "local types", scoped in WKT or an external type.

Reply:
Aside from WKT and external type records, we are also trying to support local types in the context of `unknown` type record?

Reply:
I don't see a way how could we and why should we do that. For `unknown` records, we should just expose `ArrayBuffer`. This would encourage using external type records instead of unknown records (which are a niche case and the NFC Forum likes to avoid it).